python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/test_asset_decorator.py

```diff
@@ -59,16 +62,12 @@ def my_sling_assets(): ...
 
 def test_runs_base_sling_config(
     csv_to_sqlite_replication_config: SlingReplicationParam,
-    path_to_test_csv: str,
     path_to_temp_sqlite_db: str,
     sqlite_connection: sqlite3.Connection,
 ):
     @sling_assets(replication_config=csv_to_sqlite_replication_config)
-    def my_sling_assets(sling: SlingResource):
-        for row in sling.replicate(
-            replication_config=csv_to_sqlite_replication_config,
-            dagster_sling_translator=DagsterSlingTranslator(),
-        ):
+    def my_sling_assets(context: AssetExecutionContext, sling: SlingResource):
+        for row in sling.replicate(context=context):
             logging.info(row)
```

Comment:
We should really be doing, and testing against the `AssetMaterialization` emitted by the result from `materialize`. We should ensure that the proper events are being created, not just user logs. [1]

```suggestion
    def my_sling_assets(context: AssetExecutionContext, sling: SlingResource):
        yield from sling.replicate(context=context)
```

Reply:
Sorry, didn't follow entirely. Do you recommend I add some additional asserts on the results, or is simply adding the `yield from` enough?

Reply:
Yeah, we should assert on the results of the Dagster computation.

Like below, we're asserting against the actual data being materialized. We should also assert that Dagster is aware that the data was materialized (e.g. asserting count of `AssetMaterialization`) or what not